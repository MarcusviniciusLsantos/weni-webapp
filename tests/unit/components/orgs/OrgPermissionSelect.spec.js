import { shallowMount, createLocalVue } from '@vue/test-utils';
import OrgPermissionSelect from '@/components/orgs/orgPermissionSelect.vue';
import i18n from '@/utils/plugins/i18n';

const localVue = createLocalVue();

describe('OrgPermissionSelect.vue', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(OrgPermissionSelect, {
      localVue,
      i18n,
      mocks: {
        $t: () => 'some specific text',
      },
      stubs: {
        UnnnicSelect: true,
      },
    });
  });

  it('renders a snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('roleOptions()', async () => {
    await wrapper.setData({
      roles: {
        1: 'view',
        2: 'contributor',
        3: 'admin',
      },
    });
    const response = wrapper.vm.roleOptions;
    expect(response).toStrictEqual(Object.keys(wrapper.vm.roles));
  });

  it('labelFor()', async () => {
    await wrapper.setData({
      roles: {
        1: 'view',
        2: 'contributor',
        3: 'admin',
      },
    });
    const response = wrapper.vm.labelFor(1);
    expect(response).toStrictEqual(wrapper.vm.$t());
  });
});
