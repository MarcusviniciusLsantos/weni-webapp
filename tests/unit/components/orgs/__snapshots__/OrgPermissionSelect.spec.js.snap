// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrgPermissionSelect.vue renders a snapshot 1`] = `
<unnnicselect-stub>
  <option value="1">
    some specific text
  </option>
  <option value="2">
    some specific text
  </option>
  <option value="3">
    some specific text
  </option>
</unnnicselect-stub>
`;
